src/gp_engine.cpp

```diff
@@ -8,25 +8,54 @@ GPEngine::GPEngine(SitePattern site_pattern, size_t gpcsp_count,
                    std::string mmap_file_path)
     : site_pattern_(std::move(site_pattern)),
       plv_count_(site_pattern_.PatternCount() + gpcsp_count),
-      mmapped_master_plv_(mmap_file_path, plv_count_ * site_pattern_.PatternCount()) {
+      mmapped_master_plv_(mmap_file_path, plv_count_ * site_pattern_.PatternCount())
+{
   Assert(plv_count_ > 0, "Zero PLV count in constructor of GPEngine.");
   plvs_ = mmapped_master_plv_.Subdivide(plv_count_);
   Assert(plvs_.size() == plv_count_,
          "Didn't get the right number of PLVs out of Subdivide.");
   Assert(plvs_.back().rows() == MmappedNucleotidePLV::base_count_ &&
-             plvs_.back().cols() == site_pattern_.PatternCount(),
-         "Didn't get the right shape of PLVs out of Subdivide.");
+       plvs_.back().cols() == site_pattern_.PatternCount(),
+   "Didn't get the right shape of PLVs out of Subdivide.");
   branch_lengths_.resize(gpcsp_count);
   log_likelihoods_.resize(gpcsp_count);
   q_.resize(gpcsp_count);
 
   auto weights = site_pattern_.GetWeights();
   site_pattern_weights_ =
-      Eigen::Map<Eigen::VectorXd, Eigen::Unaligned>(weights.data(), weights.size());
+    Eigen::Map<Eigen::VectorXd, Eigen::Unaligned>(weights.data(), weights.size());
 
   InitializePLVsWithSitePatterns();
 }
 
+GPEngine::GPEngine(SitePattern site_pattern,
```

Comment:
Why do we have these two very similar constructors?

Reply:
One of them was written by you and it is still called in the test https://github.com/phylovi/libsbn/blob/ee74caeb62a570f739657be4691892543b63d3dc/src/gp_engine.hpp#L148.

Should I remove the constructor even if it means to call the constructor without correct arguments?

Reply:
Done.